app/pages/project/affinity/AntiAffinityGroupPage.tsx

```diff
@@ -0,0 +1,134 @@
+import type { LoaderFunctionArgs } from 'react-router'
+
+import { Affinity24Icon } from '@oxide/design-system/icons/react'
+
+import {
+  apiq,
+  getListQFn,
+  queryClient,
+  usePrefetchedQuery,
+  type AntiAffinityGroupMember,
+} from '~/api'
+import { makeCrumb } from '~/hooks/use-crumbs'
+import {
+  getAntiAffinityGroupSelector,
+  useAntiAffinityGroupSelector,
+  useProjectSelector,
+} from '~/hooks/use-params'
+import { makeLinkCell } from '~/table/cells/LinkCell'
+import { Columns } from '~/table/columns/common'
+import { Table } from '~/table/Table'
+import { Badge } from '~/ui/lib/Badge'
+import { CardBlock } from '~/ui/lib/CardBlock'
+import { EmptyMessage } from '~/ui/lib/EmptyMessage'
+import { PropertiesTable } from '~/ui/lib/PropertiesTable'
+import { TableEmptyBox } from '~/ui/lib/Table'
+import { ALL_ISH } from '~/util/consts'
+import { pb } from '~/util/path-builder'
+import type * as PP from '~/util/path-params'
+
+import { AffinityPageHeader } from './AffinityPage'
+
+export const handle = makeCrumb(
+  (p) => p.antiAffinityGroup!,
+  (p) => pb.antiAffinityGroup(getAntiAffinityGroupSelector(p))
+)
+
+const colHelper = createColumnHelper<AntiAffinityGroupMember>()
+
+const antiAffinityGroupView = ({ antiAffinityGroup, project }: PP.AntiAffinityGroup) =>
+  apiq('antiAffinityGroupView', { path: { antiAffinityGroup }, query: { project } })
+const memberList = ({ antiAffinityGroup, project }: PP.AntiAffinityGroup) =>
+  getListQFn('antiAffinityGroupMemberList', {
+    path: { antiAffinityGroup },
+    // member limit in DB is currently 32, so pagination isn't needed
+    query: { project, limit: ALL_ISH },
+  })
+
+export async function clientLoader({ params }: LoaderFunctionArgs) {
+  const { antiAffinityGroup, project } = getAntiAffinityGroupSelector(params)
+  await Promise.all([
+    queryClient.fetchQuery(antiAffinityGroupView({ antiAffinityGroup, project })),
+    queryClient.fetchQuery(memberList({ antiAffinityGroup, project }).optionsFn()),
+  ])
+  return null
+}
+
+const AntiAffinityGroupMemberEmptyState = () => (
+  <TableEmptyBox>
+    <EmptyMessage
+      icon={<Affinity24Icon />}
+      title="No anti-affinity group members"
+      body="Add a new anti-affinity group member to see it here"
+      buttonText="Add anti-affinity group member"
+      // TODO: this should open the AntiAffinityGroupd edit modal
+      buttonTo={pb.antiAffinityGroupNew(useProjectSelector())}
+    />
+  </TableEmptyBox>
+)
+
+export default function AntiAffinityPage() {
+  const { antiAffinityGroup, project } = useAntiAffinityGroupSelector()
+  const { data: group } = usePrefetchedQuery(
+    antiAffinityGroupView({ antiAffinityGroup, project })
+  )
+  const { id, name, description, policy, timeCreated } = group
+  const { data: members } = usePrefetchedQuery(
+    memberList({ antiAffinityGroup, project }).optionsFn()
+  )
+  // TODO: Run an ALL_ISH query to get total number of members
+  const membersCount = members?.items.length ?? 0
```

Comment:
`members` is always defined because of the prefetch, so no need for `?.` or fallback. That got me interested in [this typescript-eslint rule](https://typescript-eslint.io/rules/no-unnecessary-condition), which does error on this line, but it also turns up 50 other spots and some of them are non-trivial to fix.